clients/src/main/java/org/apache/kafka/clients/admin/KafkaAdminClient.java

```
@@ -4321,6 +4328,94 @@ void handleFailure(Throwable throwable) {
+                        log.debug(msg);
+                        throw new UnknownServerException(msg);
+                    }
+                    DescribeQuorumResponseData.TopicData topic = quorumResponse.data().topics().get(0);
+                    if (!topic.topicName().equals(METADATA_TOPIC_NAME)) {
+                        String msg = String.format("DescribeMetadataQuorum received a topic with name {} when {} was expected",
+                                topic.topicName(), METADATA_TOPIC_NAME);
+                        log.debug(msg);
+                        throw new UnknownServerException(msg);
+                    }
+                    if (topic.partitions().size() > 1) {
+                        String msg = String.format("DescribeMetadataQuorum received a topic {} with {} partitions when 1 was expected",
+                                topic.topicName(), topic.partitions().size());
+                        log.debug(msg);
+                        throw new UnknownServerException(msg);
+                    }
+                    DescribeQuorumResponseData.PartitionData partition = topic.partitions().get(0);
+                    if (partition.partitionIndex() != METADATA_TOPIC_PARTITION.partition()) {
+                        String msg = String.format("DescribeMetadataQuorum received a single partition with index {} when {} was expected",
+                                partition.partitionIndex(), METADATA_TOPIC_PARTITION.partition());
+                        log.debug(msg);
+                        throw new UnknownServerException(msg);
+                    }
+                    if (partition.errorCode() != Errors.NONE.code()) {
+                        throw Errors.forCode(partition.errorCode()).exception();
+                    }
+                    future.complete(createQuorumResult(partition));
+                } catch (RuntimeException e) {
+                    throw e;
```

Is the intent to retry here? For some errors, such as auth failures, we probably would rather fail fast.

---

The reason I added this block is because I noticed a gradle warning which suggested that (with the addition of the general Exception catch block), some runtime exceptions might get hidden. A little reading on this suggested that a best practice is to catch and re-throw runtime exceptions. I guess your comment here is that it is best to just do the same thing with runtime exceptions as we are doing with other exceptions and complete the future with an error. Am I understanding that correctly?